xarray/core/groupby.py

```diff
@@ -382,7 +382,15 @@ def fillna(self, value):
         Dataset.fillna
         DataArray.fillna
         """
-        return self._fillna(value)
+        def _yield_applied(this, other):
```

Comment:
So, if I replace the whole method with just `return apply_ufunc(ops.fillna, self, value)`, I'd fail line 2507 of `test_dataset.py`, where `self.assertEqual(actual.attrs, ds.attrs)` fails, because the `attrs` aren't copied over.